### ISSUE TITLE
Add Self-Conditioning Refinement (AlphaFold-style)

### ISSUE THREAD

**Issue by AthomsG** [AthomsG/PanCake#6]
# Add Self-Conditioning Refinement (AlphaFold-style)

Add a refinement step on top of the current simple MLP to capture gene–gene dependencies.

## Idea
- **Pass 1**: X → MLP → logits/probs for 10k genes.
- Build a context vector from first-pass predictions (stop-grad).
- **Pass 2**: concat [MLP hidden, context] → small refiner → refined logits/probs.
- Train with BCE on refined outputs only.

## Tasks
- [ ] Modularize and implement refinement in `src/train-models.py` (add a `RefinementHead` block).  
- [ ] Add CLI flag `--model refinement` with relevant hyperparams.  
- [ ] Update `train-models.sh` with an example command.  
- [ ] Evaluate vs baseline (plain MLP) using AUC and precision@k.

**Comment by AthomsG** [comment]
Not really improving. Also interpretability would be complicated

<img width="1410" height="990" alt="Image" src="https://github.com/user-attachments/assets/3399d97e-3d0e-4dc1-b322-5d8e7cc9ca07" />
